aws/ec2metadata/api.go

```diff
@@ -10,6 +14,30 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/fullsailor/pkcs7"
+)
+
+const (
+	// AWSCertificatePem is the official public certificate for AWS
+	// copied from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+	AWSCertificatePem = `-----BEGIN CERTIFICATE-----
```

Comment:
The certificates are not consistent across AWS partitions. For example AWS uses this certificate, and AWS GovCloud (US) uses a different certificate. I think each partition uses a separate certificate.

Reply:
Yes that's true. Since currently AWS doesn't provide an API for requesting its public cert, perhaps we can add SDK user to supply the cert somewhere if they want to check the signature.